### ISSUE TITLE
preLaunchTask base path not consistent

### ISSUE THREAD

**Issue by synaptek** [microsoft/vscode#75084]
Issue Type: <b>Bug</b>

This is new in release 1.35.0.  When a VSCode workspace is launched from a Windows shortcut it calculates the task path relative to the VSCode install path and when a workspace is loaded manually in a running VSCode instance the path is calculated relative the workspace path.  

This now makes it impossible to have a consistent preLaunchTask configuration that works in either case.

VS Code version: Code 1.35.0 (553cfb2c2205db5f15f3ee8395bbd5cf066d357d, 2019-06-04T01:17:12.481Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-7900X CPU @ 3.30GHz (20 x 3312)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|127.68GB (120.61GB free)|
|Process Argv|E:\Projects\EdForge\ISP_TUTOR\EFTutorBuilder\TutorBuilder\TutorBuilder.code-workspace|
|Screen Reader|no|
|VM|50%|
</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-color|ans|0.4.5
dart-code|Dar|3.1.0
tslint|eg2|1.0.43
live-html-previewer|hdg|0.3.0
vscode-json5|mrm|1.0.0
cpptools|ms-|0.23.1
vscode-code-outline|pat|0.2.1
java|red|0.46.0
vscodeintellicode|Vis|1.1.6
vscode-java-debug|vsc|0.19.0
vscode-java-dependency|vsc|0.4.0
vscode-java-pack|vsc|0.7.1
vscode-java-test|vsc|0.17.0
vscode-maven|vsc|0.17.0
linecount|yyc|0.1.7

</details>

**Comment by egamma** [comment]
@alexr00 did your recent change backward compatibility?

**Comment by alexr00** [comment]
@synaptek, I tried a couple different shortcuts:
`"C:\Users\alros\AppData\Local\Programs\Microsoft VS Code Insiders\Code - Insiders.exe" C:\Users\alros\repos\Microsoft\vscode-extension-samples\helloworld-sample`
and
`"C:\Program Files\Microsoft VS Code\Code.exe" C:\Users\alros\repos\Microsoft\vscode-extension-samples\helloworld-sample`
With both of them, I was able to launch from the shortcut then use a launch configuration with a `preLaunchTask`. What does your shortcut look like?

**Comment by synaptek** [comment]
@alexr00 - It is when you open the workspace by double clicking the xxx.code-workspace (file type).  Sorry I shouldn't have said shortcut.  

On my machine this can be readily seen by selecting Terminal/Run Task... in VSCode.  It will display a list of VSCode install path relative npm/tsc tasks when the xxx.code-workspace was double clicked to open the workspace, while opening the workspace in a running instance of VSCode (File Open Workspace...) will show a list of Workspace relative npm/tsc tasks.  This results in errors when a debug launch task attempts to use a preLaunchTask that is configured one way or the other depending on how the workspace was loaded. Sorry for any confusion.

**Comment by synaptek** [comment]
@alexr00 - I just noticed that when the workspace is opened by double click the Task List base paths are actually inconsistent... i.e. some are VSCode relative and some are workspace relative.  When open using File Open Workspace they are consistent and are all workspace relative.

**Comment by alexr00** [comment]
@synaptek my new steps:
1. Open a folder that has some tasks and a debug config with a prelaunch task.
2. Save workspace file to my desktop.
3. Close instance of VS Code
4. Open the workspace and see that VS Code is launched.
5. Try to run the debug config. 
6. preLaunch task is successful and debug works.

Here is the project a tried: https://github.com/microsoft/vscode-extension-samples/tree/master/helloworld-sample

Can you make repro steps that use that project, or another simple project that I can clone to reproduce?

**Comment by synaptek** [comment]
@alexr00  - I have created a repo to reproduce this Task issue - 

Please see: 
https://github.com/EdForge/VSCODE_TEST

I have tested this on multiple machines with various install paths.

Thx
Kevin

**Comment by alexr00** [comment]
I can repro, thank you for the example and the clear steps!

**Comment by alexr00** [comment]
@synaptek, the bug I found using your repo is specific to typescript tasks and I have made a fix for it. It isn't a problem I'd expect to see with other built-in contributed tasks though. Are you seeing it with any other task types?

**Comment by RMacfarlane** [comment]
I still see strange paths with the latest insiders:
![Screen Shot 2019-06-26 at 4 13 06 PM](https://user-images.githubusercontent.com/3672607/60221159-711a2580-982d-11e9-9b02-4c429419a63f.png)

**Comment by synaptek** [comment]
@alexr00 - Just in stalled following insider build and the fault is still there.  If this build isn't expected to have the fix please let me know the version I can expect it to be in so I can re-test.

Thx!

Version: 1.36.0-insider (user setup)
Commit: b7265738444f1eee05762341b4361ee6452a4154
Date: 2019-06-27T05:22:06.378Z
Electron: 4.2.5
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.17134

**Comment by RMacfarlane** [comment]
@synaptek Sorry, I should have removed my "verified" tag earlier. I added it this morning and then realized that this change hadn't made it into the latest build, and was waiting for us to release a new build. I added the tag preemptively to prevent collisions on our team of other people trying to verify it at the same time. We should have a new insiders build to test within the next couple of hours.
